syntax = "proto3";

package github.v1;

import "google/api/annotations.proto";
import "google/api/httpbody.proto";
import "resemble/v1alpha1/options.proto";

////////////////////////////////////////////////////////////////////////

service RepositoryInterface {
  rpc Attach(AttachRequest) returns (AttachResponse) {
    option (resemble.v1alpha1.method) = {
      writer: {
        constructor: {},
      },
    };
  }
}

////////////////////////////////////////////////////////////////////////

message Repository {
  option (.resemble.v1alpha1.state) = {
  };

  string org = 1;
  string repo = 2;
}

////////////////////////////////////////////////////////////////////////

message AttachRequest {
  string org = 1;
  string repo = 2;
}

message AttachResponse {}

////////////////////////////////////////////////////////////////////////

service Http {
  // Serve a web dashboard. Uses `HttpBody` to return arbitrary content (HTML,
  // JS) to callers; see `httpbody.proto` for more details.
  rpc Webhook(WebhookRequest) returns (google.api.HttpBody) {
    option (google.api.http) = {
      post: "/github/v1/repository/webhook"
      body: "*"
    };
  }
}

////////////////////////////////////////////////////////////////////////

message WebhookRequest {
  string data = 1;
}

////////////////////////////////////////////////////////////////////////
